bookkeeper-server/src/main/java/org/apache/bookkeeper/client/ITopologyAwareEnsemblePlacementPolicy.java

```diff
@@ -122,6 +122,29 @@ T selectFromNetworkLocation(String networkLoc,
                                 boolean fallbackToRandom)
             throws BKException.BKNotEnoughBookiesException;
 
+    /**
+     * Select a node from a given network location.
+     *
+     * @param networkLoc
+     *          network location
+     * @param excludeBookies
+     *          exclude bookies set
+     * @param predicate
+     *          predicate to apply
+     * @param ensemble
+     *          ensemble
+     * @param fallbackToRandom
+     *          fallbackToRandom
+     * @return the selected bookie.
+     * @throws BKException.BKNotEnoughBookiesException
+     */
+    T selectFromNetworkLocationWithPickCommonRackFirst(String networkLoc,
```

Comment:
Need a default implementation.

Reply:
Followed the method `selectFromNetworkLocation`; it's not a default method. In the `ZoneawareEnsemblePlacementPolicyImpl`, it implements the method and throw exception.
https://github.com/apache/bookkeeper/blob/405e72acf42bb1104296447ea8840d805094c787/bookkeeper-server/src/main/java/org/apache/bookkeeper/client/ZoneawareEnsemblePlacementPolicyImpl.java#L303-L309